docs/multiplatform.md

```diff
@@ -171,19 +176,14 @@ The only difference is for projects that generate code into each KMP target sour
 kotlin {
     // add your project's targets here like in the snippet above
 
-    configureCommonMainKsp()
+    commonMain {
+        kotlin.srcDir("build/generated/ksp/metadata/commonMain/kotlin")
+    }
 }
 
-@OptIn(ExternalVariantApi::class)
-fun KotlinMultiplatformExtension.configureCommonMainKsp() {
-  sourceSets.named("commonMain").configure {
-    kotlin.srcDir("build/generated/ksp/metadata/commonMain/kotlin")
-  }
-
-  project.tasks.withType(KotlinCompilationTask::class.java).configureEach {
-    if(name != "kspCommonMainKotlinMetadata") {
-      dependsOn("kspCommonMainKotlinMetadata")
+tasks.withType<KspAATask>().configureEach {
```

Comment:
Might be worth leaving a comment saying this is for KSP2, and that projects using KSP1 should use `KotlinCompilationTask`.